File: tests/fixtures/chunker_samples/go/sample.go
// Sample Go file for tree-sitter chunker tests.
package sample

import "fmt"

type User struct {
	ID   int
	Name string
}

func FormatMessage(name string) string {
	return fmt.Sprintf("hello, %s!", name)
}

func (u *User) Display() string {
	return fmt.Sprintf("%d: %s", u.ID, u.Name)
}

File: tests/fixtures/chunker_samples/rust/sample.rs
// Sample Rust file for tree-sitter chunker tests.

pub struct User {
    pub id: u32,
    pub name: String,
}

pub enum Greeting {
    Casual,
    Formal,
}

pub fn format_message(name: &str) -> String {
    format!("hello, {}!", name)
}

impl User {
    pub fn display(&self) -> String {
        format!("{}: {}", self.id, self.name)
    }
}
